core/comp_ui.py

```diff
@@ -44,12 +45,15 @@ def _PromptLen(prompt_str):
     elif c == '\x02':
       escaped = False
     elif not escaped:
-      display_str += c
+      # mycpp: rewrite of +=
+      display_str = display_str + c
   last_line = display_str.split('\n')[-1]
   try:
     width = libc.wcswidth(last_line)
   # en_US.UTF-8 locale missing, just return the number of bytes
-  except (SystemError, UnicodeError):
+  except UnicodeError:
+    return len(display_str)
+  except OSError:
```

Comment:
Please remove OSError here, since I think we should have a separate path for allocation failure

An OSError should only be for things Python raises OSError for

(It actually raises `MemoryError` for out of memory, but I'm not sure if we should do that as well)